docs/proposals/25-namespaced-cloud-profiles.md

```diff
@@ -0,0 +1,338 @@
+
+A use case could be defined where a `NamespacedCloudProfile` might want to be shared across multiple projects and not just be used within the project it was created in. Especially when taking a broader view of Gardeners development with Private Seeds and Cloud in Country, this feature is probably going to be necessary at some point.
+
+This GEP already modifies the `cloudProfile` field in the `Shoot`s spec. To implement cross-project sharing, a `namespace` field could be added to the `cloudProfile` field. It specifies in which project/namespace the selected `CloudProfile` is. Checking if a user is allowed to select the specified `CloudProfile` can be handled in multiple ways and should be specified more concretely when this feature is to be implemented.
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Shoot
+metadata:
+  name: my-shoot
+  namespace: project-xyz
+  ...
+spec:
+  ...
+  cloudProfile:
+    kind: NamespacedCloudProfile
+    name: aws-profile-xyz
+    namespace: other-project-abc
+  ...
+```
+
+### Multi-Level inheritance
+
+A `NamespacedCloudProfile` could inherit from a `NamespacedCloudProfile` that already inherits from a `CloudProfile`. This would enable reusing `NamespacedCloudProfile`s and would aid in deduplication within fields of `NamespacedCloudProfiles`. This feature can be implemented easily but is also excluded for now to stick to the defined use cases.
+
+When enabling multi-level inheritance, the `NamespacedCloudProfile`s `parent` field should also be adjusted to allow for a `NamespacedCloudProfile` as a parent.
+
+```yaml
+parent:
+  kind: <NamespacedCloudProfile | CloudProfile>
+  name: <objects name>
+```
+
+In combination with [cross-project sharing](#cross-project-sharing), the `parent` field should also allow for a namespace to be defined with the same reasoning as in [cross-project sharing](#cross-project-sharing).
+
+```yaml
+parent:
+  kind: <NamespacedCloudProfile | CloudProfile>
+  name: <objects name>
+  namespace: <(optional) objects namespace if outside project namespace>
+```
+
+### `NamespacedCloudProfile`s replacing regular `CloudProfile`s
+
+In the future, once `NamespacedCloudProfiles` have established themselves and found good use in the Gardener landscape and amongst its users, they could replace regular `CloudProfile`s entirely. For this, they would have to include all fields of a `CloudProfile`. Replacing `CloudProfile`s with `NamespacedCloudProfile`s has several benefits. Firstly, Gardener maintainers don't have to maintain, test and operate two almost identical objects. Secondly, it would allow for the central "Gardener provided" `CloudProfile`s to use inheritance. This could enable deduplication in our central `CloudProfiles` as common values do not have to be copied from one `CloudProfile` to another `CloudProfile`. Additionally, it could enable both cloud, as well as on-prem infrastructures to only be visible to defined projects and not to every landscape user.
```

Comment:
I am not sure whether that would be sensible. Then the project admins would have the burden to find out and maintain their namespaced cloud profiles with all their values for the hyperscalers, no?

I was more thinking that also our cluster cloud profiles support inheritance, i.e. something like:
cluster root cloud profile (with Kubernetes versions for instance) <- cluster hyperscaler cloud profiles (like today) <- namespaced cloud profiles/hierarchy (optional)

Anyway, that can be done later as you said. Thank you.